eventhubs/data-plane/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/ManagedIdentityTokenProvider.java

```diff
@@ -0,0 +1,26 @@
+package com.microsoft.azure.eventhubs;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+
+import com.microsoft.azure.credentials.MSICredentials;
+import com.microsoft.azure.eventhubs.impl.ClientConstants;
+
+public class ManagedIdentityTokenProvider implements ITokenProvider {
+	final static MSICredentials credentials = new MSICredentials();
```

Comment:
That won't support:
- Webapp as a whole, you need a second class: https://github.com/Azure/autorest-clientruntime-for-java/blob/a55f87f3cc3a68742a2ac94c031c6d715965a9c2/azure-client-authentication/src/main/java/com/microsoft/azure/credentials/AppServiceMSICredentials.java
- User Assigned Identity, you need to be able to provide a client_id / object_id for that